tests/testthat/test-plot_step_ahead_model_output.R

```diff
@@ -0,0 +1,231 @@
+  testthat::expect_false(plot_test$x$layoutAttrs[[1]]$showlegend)
+  testthat::expect_gt(length(unique(purrr::map(plot_test$x$data, "yaxis"))), 1)
+  testthat::expect_gt(length(unique(purrr::map(plot_test$x$data, "xaxis"))), 1)
+  testthat::expect_equal(
+    length(purrr::discard(purrr::map(plot_test$x$data[
+      purrr::map(plot_test$x$data, "name") == "ground truth"], "x"), is.null)),
+    1
+  )
+
+  plot_test <- plot_step_ahead_model_output(
+    projection_data, truth_data_us, top_layer = "truth",
+    facet = "model_id", fill_by = "scenario_id", facet_title = "bottom right")
+  testthat::expect_equal(tail(plot_test$x$data, 1)[[1]]$name, "ground truth")
+  testthat::expect_equal(
+    unique(unlist(purrr::map(plot_test$x$layout$annotations, "xanchor"))),
+    "right")
+  testthat::expect_equal(
+    unique(unlist(purrr::map(plot_test$x$layout$annotations, "yanchor"))),
+    "bottom")
+  testthat::expect_in(
+    unlist(purrr::map(plot_test$x$layout$annotations, "text")),
+    unique(projection_data$model_id)
+  )
+  testthat::expect_in(
+    unique(unlist(purrr::map(plot_test$x$attrs, "color")))  ,
+    unique(projection_data$scenario_id)
+  )
+
+  plot_test <- plot_step_ahead_model_output(
+    projection_data, truth_data_us, facet = "model_id", title = "My Plot",
+    facet_scales = "free_y", facet_nrow = 3, fill_transparency = 0.75,
+    use_median_as_point = TRUE)
+  testthat::expect_gt(length(unique(purrr::map(plot_test$x$data, "yaxis"))), 1)
+  testthat::expect_equal(length(unique(purrr::map(plot_test$x$data, "xaxis"))),
+                         1)
+  testthat::expect_equal(plot_test$x$layoutAttrs[[2]]$title, "My Plot")
+  testthat::expect_in(
+    unlist(purrr::map(plot_test$x$layout$annotations, "text")),
+    unique(projection_data$model_id)
+  )
+  testthat::expect_in(
+    unique(unlist(purrr::map(plot_test$x$attrs, "color")))  ,
+    unique(projection_data$model_id)
+  )
+  testthat::expect_equal(tail(plot_test$x$data, 1)[[1]]$opacity, 0.75)
+
+  plot_test <- plot_step_ahead_model_output(
+    projection_data, truth_data_us, pal_color = NULL, one_color = "orange",
+    ens_color = "black", ens_name = "hub-ensemble", intervals = NULL,
+    use_median_as_point = TRUE, facet = "scenario_id")
+  testthat::expect_equal(
+    unique(unlist(purrr::map(plot_test$x$data[purrr::map(
+      plot_test$x$data, "name") == "hub-ensemble"], "line"))["color"]), "black")
+  testthat::expect_equal(
+    unique(unlist(purrr::map(plot_test$x$data[purrr::map(
+      plot_test$x$data, "name") == "HUBuni-simexamp"], "line"))["color"]),
+    "rgba(255,165,0,1)")
+
+  plot_test <- plot_step_ahead_model_output(
+    projection_data_A_us, truth_data_us, intervals = 0.9, ens_color = "black",
+    ens_name = "hub-ensemble", facet = "model_id")
+  testthat::expect_equal(
+    strsplit(strsplit(tail(purrr::map(
+      plot_test$x$attrs, "hovertext"), 1)[[1]], "<br>")[[1]][2],
+      "Intervals")[[1]][1], "90%")
+
+  })
```

Comment:
Just to note that snapshots of graphical output can be tested using [`vdiffr`](https://vdiffr.r-lib.org/) package

Reply:
Thanks I was looking into it and thinking to implement it in the next version